src/convert/transformers/staticResourceMetadataTransformer.ts

```diff
@@ -128,54 +120,61 @@ export class StaticResourceMetadataTransformer extends BaseMetadataTransformer {
-  }
-
   /**
    * "Expanded" refers to a component whose content file is a zip file, and its current
    * state is unzipped.
    */
-  private async componentIsExpandedArchive(component: SourceComponent): Promise<boolean> {
-    const { content, tree } = component;
-    if (tree.isDirectory(content)) {
-      const contentType = await this.getContentType(component);
-      if (StaticResourceMetadataTransformer.ARCHIVE_MIME_TYPES.has(contentType)) {
-        return true;
-      }
-      throw new SfError(
-        messages.getMessage('error_static_resource_expected_archive_type', [contentType, component.name]),
-        'LibraryError'
-      );
-    }
-    return false;
-  }
+}
 
-  private async getContentType(component: SourceComponent): Promise<string> {
-    const resource = (await component.parseXml()).StaticResource as JsonMap;
+const DEFAULT_CONTENT_TYPE = 'application/octet-stream';
+const FALLBACK_TYPE_MAP = new Map<string, string>([
+  ['text/javascript', 'js'],
+  ['application/x-javascript', 'js'],
+  ['application/x-zip-compressed', 'zip'],
+  ['text/x-haml', 'haml'],
+  ['image/x-png', 'png'],
+  ['text/xml', 'xml'],
+]);
 
-    if (!resource || !Object.prototype.hasOwnProperty.call(resource, 'contentType')) {
-      throw new SfError(
-        messages.getMessage('error_static_resource_missing_resource_file', [join('staticresources', component.name)]),
-        'LibraryError'
-      );
-    }
+const getContentType = async (component: SourceComponent): Promise<string> => {
+  const resource = (await component.parseXml()).StaticResource as JsonMap;
 
-    return resource.contentType as string;
+  if (!resource || !Object.prototype.hasOwnProperty.call(resource, 'contentType')) {
+    throw new SfError(
+      messages.getMessage('error_static_resource_missing_resource_file', [join('staticresources', component.name)]),
+      'LibraryError'
+    );
   }
 
-  private getExtensionFromType(contentType: string): string {
-    // return registered ext, fallback, or the default (application/octet-stream -> bin)
-    return (
-      getExtension(contentType) ||
-      StaticResourceMetadataTransformer.FALLBACK_TYPE_MAP.get(contentType) ||
-      getExtension(StaticResourceMetadataTransformer.DEFAULT_CONTENT_TYPE)
+  return resource.contentType as string;
+};
+
+const getBaseContentPath = (component: SourceComponent, mergeWith?: SourceComponent): SourcePath => {
+  const baseContentPath = mergeWith?.content || component.getPackageRelativePath(component.content, 'source');
+  return join(dirname(baseContentPath), baseName(baseContentPath));
+};
+
+const getExtensionFromType = (contentType: string): string =>
+  // return registered ext, fallback, or the default (application/octet-stream -> bin)
+  getExtension(contentType) ?? FALLBACK_TYPE_MAP.get(contentType) ?? getExtension(DEFAULT_CONTENT_TYPE);
+
+const componentIsExpandedArchive = async (component: SourceComponent): Promise<boolean> => {
```

Comment:
just curious when you refactor these are you using an IDE to pull the functions out?

Reply:
nope, there's not a good action for that that I've found so far.